snapshot/exporter/sftp/sftp_test.go

```diff
@@ -0,0 +1,122 @@
+package sftp
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/PlakarKorp/plakar/appcontext"
+	"github.com/PlakarKorp/plakar/objects"
+	ptesting "github.com/PlakarKorp/plakar/testing"
+	"golang.org/x/crypto/ssh"
+)
+
+func TestExporter(t *testing.T) {
+	// Generate a keypair for the test
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatalf("Failed to generate private key: %v", err)
+	}
+	privBytes := x509.MarshalPKCS1PrivateKey(priv)
+	privPEM := pem.EncodeToMemory(&pem.Block{Type: "RSA PRIVATE KEY", Bytes: privBytes})
+	pub, err := ssh.NewPublicKey(&priv.PublicKey)
+	if err != nil {
+		t.Fatalf("Failed to generate public key: %v", err)
+	}
+
+	// Write private key to a temp file
+	keyFile, err := os.CreateTemp("", "sftp-exporter-key-*.pem")
+	if err != nil {
+		t.Fatalf("Failed to create temp key file: %v", err)
+	}
+	defer os.Remove(keyFile.Name())
+	if _, err := keyFile.Write(privPEM); err != nil {
+		t.Fatalf("Failed to write private key: %v", err)
+	}
+	keyFile.Close()
```

Comment:
can you please move this into NewMockSFTPServer?  Just in case we have to add further tests, we'd have less copy-pasta.

Reply:
good idea, done in https://github.com/PlakarKorp/plakar/pull/988/commits/0284128e961fb68ac17b5b89624577f5d6121f8f

Reply:
lovely!